src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java

```diff
@@ -186,10 +210,19 @@ private void doMapAuthRecord(ExtMap input, ExtMap output) throws Exception {
     private Format createFormat(Properties props, String prefix) {
         Format ret = null;
         String type = props.getProperty(prefix + "type", "void");
+        String newcase = props.getProperty(prefix + "case", "").toLowerCase();
+        ChangeCase changecase;
+        if("lower".equals(newcase)) {
+            changecase = ChangeCase.TOLOWER;
+        } else if ("upper".equals(newcase)){
+            changecase = ChangeCase.TOUPPER;
+        } else {
+            changecase = ChangeCase.KEEP;
+        }
```

Comment:
Could you please move the conversion logic into the `ChangeCase` to `public static ChangeCase valueOf(String value)` method?

Reply:
Is throwing IllegalArgumentException ok for you ? Or does it needs to be catched ?